### DISCUSSION TITLE
Memcode SM2 formula

### DISCUSSION THREAD

**Discussion by lakesare** [lakesare/memcode#142] [Ideas]
@mustafa0x built a nice interface for playing with our SM2 algorithm: https://svelte.dev/repl/810665a44c904c17876fea2fc36fed8d?version=3.46.2

> I wrote a quick test page for getNextScore, and seemed to be a good choice, but I did notice something. The difference between grade 5 and 4 seems to be very minor. Also, 1-3 seem to be fail grades, which seems to be a bit much.
([His comments](https://github.com/lakesare/memcode/pull/138#issuecomment-1013907462)).

As you see, 4 is barely different from 5. It's especially problematic that after grade 4 you'll be scheduled to review the flashcard in 44 days - it should be much sooner.
I notice I do use 4 as a very mild form of punishment when I'm reviewing the flashcards myself (meaning the user does get adjusted), however it would be better if we reviewed the flaschard again in, e.g., 1 day, then in 10 days, and then fully recovered our progress to the previous levels.

Grade 5             |  Grade 4
:-------------------------:|:-------------------------:
<img width="300" mar alt="image" src="https://user-images.githubusercontent.com/7578559/149733225-71185c8d-658c-4aa7-baf2-6e667b3a05b7.png">  |  <img width="300" alt="image" src="https://user-images.githubusercontent.com/7578559/149732672-204af7ce-4225-4101-befb-9a8074236a9a.png">

**Comment by lakesare** [comment]
Ideally we would manually specify the desired "scheduled to (days)", and then perform regression analysis, could be a good BSc thesis for someone (@kgashok, if someone's searching for a topic).

**Comment by kgashok** [comment]
@Pavithratrdev Please discuss with our faculty.

**Reply by Pavithratrdev** [reply]
Sure Sir

**Comment by mustafa0x** [comment]
Thanks for opening this up for discussion. For now I'm using a very simple exponential scheduler (number of times seen ^ n), but a more robust algorithm would be very nice indeed.
